docs/Training-Configuration-File.md

```diff
@@ -59,6 +59,8 @@ the `trainer` setting above).
 | :---------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `hyperparameters -> beta`      | (default = `5.0e-3`) Strength of the entropy regularization, which makes the policy "more random." This ensures that agents properly explore the action space during training. Increasing this will ensure more random actions are taken. This should be adjusted such that the entropy (measurable from TensorBoard) slowly decreases alongside increases in reward. If entropy drops too quickly, increase beta. If entropy drops too slowly, decrease `beta`. <br><br>Typical range: `1e-4` - `1e-2`                                                                                                                                                                     |
 | `hyperparameters -> epsilon`   | (default = `0.2`) Influences how rapidly the policy can evolve during training. Corresponds to the acceptable threshold of divergence between the old and new policies during gradient descent updating. Setting this value small will result in more stable updates, but will also slow the training process. <br><br>Typical range: `0.1` - `0.3`                                                                                                                                                                                                                                                                                                                      |
+| `hyperparameters -> beta_schedule` | (default = `linear`) Determines how beta changes over time. <br><br>`linear` decays beta linearly, reaching 0 at max_steps, while `constant` keeps beta constant for the entire training run.                                                                                                           |
+| `hyperparameters -> epsilon_schedule` | (default = `linear`) Determines how epsilon changes over time (PPO only). <br><br>`linear` decays epsilon linearly, reaching 0 at max_steps, while `constant` keeps the epsilon constant for the entire training run.
```

Comment:
After second thought, I think the default value should be the same as the learning_rate_schedule. These default values are changing the behavior of old config yaml and users will have to update **all** their yaml files. Although it has not been released as 1.0 yet, this could cause a huge confusion and troubles if people didn't notice this change.

I'd propose here we set the default value as whatever the learning_rate_schedule is, and revert all the changes to the config files. And create a ticket to remind ourselves to change this if we ever do trainer major release. If you strongly believe we should keep these default values we can discuss offline and in that case you should add it to the migration guide.

Reply:
That's what I did. I set the default to linear, which is what the default LR schedule. I also went in and added it to the config files to match whatever the LR schedule was set to, to make sure that there was no change to training. Not sure what else to do.

Reply:
oh I saw you set something to constant originally.

But if a user has a config file that specified `learning_rate_schedule: constant`, they will get different behavior using the old trainer and new trainer right?

Reply:
I'm not sure there's a great way to use the value of one config for others if not specified using attrs i.e. default to lr schedule if beta/epsilon are not scheduled. It's possible that this can be done in a validator/structure method.

Another benefit of this would be that we don't need to add it to all of the configs.